discovery_engine_core/bindings/src/types/init_config.rs

```diff
@@ -213,6 +213,28 @@ pub unsafe extern "C" fn init_config_place_of_de_config(
     unsafe { addr_of_mut!((*place).de_config) }
 }
 
+/// Returns a pointer to the `data_dir` field of a configuration.
+///
+/// # Safety
+///
+/// The pointer must point to a valid [`InitConfig`] memory object,
+/// it might be uninitialized.
+#[no_mangle]
+pub unsafe extern "C" fn init_config_place_of_data_dir(place: *mut InitConfig) -> *mut String {
+    unsafe { addr_of_mut!((*place).data_dir) }
+}
+
+/// Returns a pointer to the `use_in_memory_db` field of a configuration.
+///
+/// # Safety
+///
+/// The pointer must point to a valid [`InitConfig`] memory object,
+/// it might be uninitialized.
+#[no_mangle]
+pub unsafe extern "C" fn init_config_place_of_use_in_memory_db(place: *mut InitConfig) -> *mut u8 {
+    unsafe { addr_of_mut!((*place).use_in_memory_db).cast() }
+}
```

Comment:
On some platforms dart ffigen doesn't handle `bool` correctly :shrug: 

This has been somewhat improve in newer versions (I'm not sure if they are released yet) but in consideration that it will soon all be removed it makes little sense to spent additional time and trying to fix it again.